src/Symfony/Component/Translation/Tests/TranslatorCacheTest.php

```diff
@@ -51,91 +52,107 @@ protected function deleteTmpDir()
         rmdir($this->tmpDir);
     }
 
-    public function testTransWithoutCaching()
+    /**
+     * @dataProvider runForDebugAndProduction
+     */
+    public function testThatACacheIsUsed($debug)
     {
-        $translator = $this->getTranslator($this->getLoader());
-        $translator->setLocale('fr');
-        $translator->setFallbackLocales(array('en', 'es', 'pt-PT', 'pt_BR', 'fr.UTF-8', 'sr@latin'));
-
-        $this->assertEquals('foo (FR)', $translator->trans('foo'));
-        $this->assertEquals('bar (EN)', $translator->trans('bar'));
-        $this->assertEquals('foobar (ES)', $translator->trans('foobar'));
-        $this->assertEquals('choice 0 (EN)', $translator->transChoice('choice', 0));
-        $this->assertEquals('no translation', $translator->trans('no translation'));
-        $this->assertEquals('foobarfoo (PT-PT)', $translator->trans('foobarfoo'));
-        $this->assertEquals('other choice 1 (PT-BR)', $translator->transChoice('other choice', 1));
-        $this->assertEquals('foobarbaz (fr.UTF-8)', $translator->trans('foobarbaz'));
-        $this->assertEquals('foobarbax (sr@latin)', $translator->trans('foobarbax'));
+        $locale = 'whatever';
+        $format = 'what-you-like';
```

Comment:
I understand your objections. However I'd like to make it as obvious as possible which values and settings matter for a test and which do not.

For example, the TranslatorTest inside FrameworkBundle on the 2.7 branch reads fixtures from disk, so the exact locale and format names are important.

Does that make sense?